include/deal.II/non_matching/mapping_info.h

```diff
@@ -226,6 +228,15 @@ namespace NonMatching
        * QSimplex::compute_affine_transformation().
        */
       bool use_global_weights;
+
+      /**
+       * During the reinit() function calls, cells are passed as
```

Comment:
You only implement this for `reinit_cells()`. This should be documented. Would it make sense for `reinit_faces()` and the other `reinit` functions?

Reply:
I have updated the documentation and added some asserts. 

> Would it make sense for reinit_faces() and the other reinit functions?

This would be great. But for that, we would also need to adjust the functions in `FEPointEvaluation`.

Reply:
> This would be great. But for that, we would also need to adjust the functions in FEPointEvaluation.

Let's keep it like this for now.